cloudsync_onedrive.py

```diff
@@ -1387,15 +1425,17 @@ def _get_validated_namespace(self, ns_id: str):
                     raise CloudNamespaceError(f"Unknown site id: {ns_id}")
                 self._fetch_drives_for_site(site)
                 drive = self.__drive_by_id.get(ns_id)
-                if not drive:
-                    # check if it is a shared drive
+                if not drive and site == self._shared_with_me and not ids.is_shared and self._is_biz:
+                    # backwards compatibility for legacy shared folder namespaces (ODB only)
                     drive = next((d for d in self._shared_with_me.drives if d.drive_id == ids.drive_id), None)
+                    if drive:
+                        name = "/".join(drive.name.split("/")[:-1])
+                        drive = Drive(name=name, id=ns_id)
                 if not drive:
                     raise CloudNamespaceError(f"Site does not contain drive: {ns_id}")
             elif ids.drive_id:
-                if ids.drive_id == self._personal_drive.drives[0].drive_id:
-                    drive = self._personal_drive.drives[0]
-                else:
+                drive = next((d for d in self._personal_drive.drives if d.drive_id == ids.drive_id), None)
```

Comment:
does this relate to onedrive personal only, or is this applicable to both business and personal?

Reply:
both.

this was a bit of unrelated cleanup, I noticed this codepath did not handle the case where a user has multiple personal drives (which is rare, but we've seen a few cases of it)